packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayer.m

```diff
@@ -421,6 +426,216 @@ - (void)setPlaybackSpeed:(double)speed error:(FlutterError *_Nullable *_Nonnull)
+        continue;
+      }
+
+      NSString *displayName = option.displayName;
+
+      NSString *languageCode = nil;
+      if (option.locale) {
+        languageCode = option.locale.languageCode;
+      }
+
+      NSString *commonMetadataTitle = nil;
+      NSArray<AVMetadataItem *> *titleItems =
+          [AVMetadataItem metadataItemsFromArray:option.commonMetadata
+                                         withKey:AVMetadataCommonKeyTitle
+                                        keySpace:AVMetadataKeySpaceCommon];
+      if (titleItems.count > 0 && titleItems.firstObject.stringValue) {
+        commonMetadataTitle = titleItems.firstObject.stringValue;
+      }
+
+      BOOL isSelected = (currentSelection == option) || [currentSelection isEqual:option];
+
+      FVPMediaSelectionAudioTrackData *trackData =
+          [FVPMediaSelectionAudioTrackData makeWithIndex:i
+                                             displayName:displayName
+                                            languageCode:languageCode
+                                              isSelected:isSelected
+                                     commonMetadataTitle:commonMetadataTitle];
+
+      [mediaSelectionTracks addObject:trackData];
+    }
+
+    // Always return media selection tracks when there's a media selection group
+    // even if all options were nil/invalid (empty array)
+    return [FVPNativeAudioTrackData makeWithAssetTracks:nil
+                                   mediaSelectionTracks:mediaSelectionTracks];
+  }
+
+  // If no media selection group or empty, try to get tracks from AVAsset (for regular video files)
+  NSArray<AVAssetTrack *> *assetAudioTracks = [asset tracksWithMediaType:AVMediaTypeAudio];
+  NSMutableArray<FVPAssetAudioTrackData *> *assetTracks = [[NSMutableArray alloc] init];
+
+  for (NSInteger i = 0; i < assetAudioTracks.count; i++) {
+    AVAssetTrack *track = assetAudioTracks[i];
+
+    // Extract metadata from the track
+    NSString *language = nil;
+    NSString *label = nil;
+
+    // Try to get language from track
+    if ([track.languageCode length] > 0) {
+      language = track.languageCode;
+    }
+
+    // Try to get label from metadata
+    for (AVMetadataItem *item in track.commonMetadata) {
+      if ([item.commonKey isEqualToString:AVMetadataCommonKeyTitle] && item.stringValue) {
+        label = item.stringValue;
+        break;
+      }
+    }
+
+    // Extract format information
+    NSNumber *bitrate = nil;
+    NSNumber *sampleRate = nil;
+    NSNumber *channelCount = nil;
+    NSString *codec = nil;
+
+    // Attempt format description parsing
+    if (track.formatDescriptions.count > 0) {
+      @try {
+        id formatDescObj = track.formatDescriptions[0];
+
+        // Validate that we have a valid format description object
+        if (formatDescObj && [formatDescObj respondsToSelector:@selector(self)]) {
+          NSString *className = NSStringFromClass([formatDescObj class]);
```

Comment:
Still, I don't understand the purpose of this if condition and code in the block. The implementation seems to be very hacky.

Reply:
You're right, this was overly defensive. The AVAssetTrack.formatDescriptions array is guaranteed by Apple's API to contain valid CMFormatDescriptionRef objects for audio tracks, so the class name string matching and respondsToSelector: checks were unnecessary.

have fixed it